pczt/src/common.rs

```diff
@@ -0,0 +1,202 @@
+    pub(crate) consensus_branch_id: u32,
+
+    /// The transaction locktime to use if no inputs specify a required locktime.
+    ///
+    /// - This is set by the Creator.
+    /// - If omitted, the fallback locktime is assumed to be 0.
+    pub(crate) fallback_lock_time: Option<u32>,
+
+    pub(crate) expiry_height: u32,
+
+    /// The [SLIP 44] coin type, indicating the network for which this transaction is
+    /// being constructed.
+    ///
+    /// This is technically information that could be determined indirectly from the
+    /// `consensus_branch_id` but is included explicitly to enable easy identification.
+    /// Note that this field is not included in the transaction and has no consensus
+    /// effect (`consensus_branch_id` fills that role).
+    ///
+    /// - This is set by the Creator.
+    /// - Roles that encode network-specific information (for example, derivation paths
+    ///   for key identification) should check against this field for correctness.
+    ///
+    /// [SLIP 44]: https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+    pub(crate) coin_type: u32,
+
+    /// A bitfield for various transaction modification flags.
+    ///
+    /// - Bit 0 is the Inputs Modifiable Flag and indicates whether inputs can be modified.
+    ///   - This is set to `true` by the Creator.
+    ///   - This is checked by the Constructor before adding inputs, and may be set to
+    ///     `false` by the Constructor.
+    ///   - This is set to `false` by the IO Finalizer if there are shielded spends or
+    ///     outputs.
+    ///   - This is set to `false` by a Signer that adds a signature that does not use
+    ///     `SIGHASH_ANYONECANPAY`.
+    ///   - The Combiner merges this bit towards `false`.
+    /// - Bit 1 is the Outputs Modifiable Flag and indicates whether outputs can be
+    ///   modified.
+    ///   - This is set to `true` by the Creator.
+    ///   - This is checked by the Constructor before adding outputs, and may be set to
+    ///     `false` by the Constructor.
+    ///   - This is set to `false` by the IO Finalizer if there are shielded spends or
+    ///     outputs.
+    ///   - This is set to `false` by a Signer that adds a signature that does not use
+    ///     `SIGHASH_NONE`.
+    ///   - The Combiner merges this bit towards `false`.
+    /// - Bit 2 is the Has `SIGHASH_SINGLE` flag and indicates whether the transaction has
+    ///   a `SIGHASH_SINGLE` signature who's input and output pairing must be preserved.
+    ///   - This is set to `false` by the Creator.
+    ///   - This is updated by a Constructor.
+    ///   - This is set to `true` by a Signer that adds a signature that uses
+    ///     `SIGHASH_SINGLE`.
+    ///   - This essentially indicates that the Constructor must iterate the transparent
+    ///     inputs to determine whether and how to add a transparent input.
+    ///   - The Combiner merges this bit towards `true`.
+    /// - Bits 3-7 must be 0.
```

Comment:
All of this would obviously need to be changed if adding `FLAG_SHIELDED_MODIFIABLE`, but I'll wait to see if I'm missing something first.

Reply:
I made changes to this documentation as part of defining Bit 7.